arboretum/common/kube_utils.py

```diff
@@ -0,0 +1,43 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2021 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common Kubernetes functions."""
+
+from requests import HTTPError
+
+
+def get_cluster_resources(session, token, resource_types, verify=True):
+    """Get resource from cluster.
+
+    :param requests.Session session: a requests.session object
+    :param str token: Bearer token
+    :param list resource_types: list of resource types. Valid values can
+      be pluralized resource types like "nodes" and "pods" or custom
+      resource API names such as "apigroup.example.com/v1/mycustom".
+    :param verify: path to a CA certificate, or True/False. Set to False to
+      skip TLS server certificate verification.  Defaults to True.
+    """
+    session.headers.update({'Authorization': f'Bearer {token}'})
+    resources = {}
+    for resource_type in resource_types:
+        base_url = 'apis' if '/' in resource_type else 'api/v1'
```

Comment:
Just a question:

If custom resource types are always going to be of the form "APIGROUP/VERSION/NAME" as you state in the README, would it be safer to implement this `if` logic that way?  I guess this is just a question of how comfortable are you that there isn't/won't be a resource type with a `/` in it that isn't a custom resource type.

Reply:
@alfinkel Do you mean this kind of logic?
```suggestion
        m = re.match(r'[^/]+/[^/]+/[^/]+',resource_type)
        base_url = 'apis' if m else 'api/v1'
```

And I realized that "custom resource" is not an appropriate term. 
There are three resource types (reference: [Kubernetes API Overview \| Kubernetes](https://kubernetes.io/docs/reference/using-api/#api-groups)).

1. core group (e.g., pod, configmap)
2. named group (e.g., apps/v1/deployment)
    - custom resource also uses named group (e.g., compliance.openshift.io/v1alpha1/compliancesuite)

I will update README.md to align this.